4exercise2improved.html

```diff
@@ -0,0 +1,60 @@
+<!DOCTYPE html>
+<html>
+
+    <head>
+        <title>Task</title>
+    
+    </head>
+<body>
+
+    <script>
+
+let calculator = {
+    
+    read() {
+        let a;
+        let b;
+        while (isNaN(a) || a === '' || a === null){
+            a = prompt ('Введите число a','');
+        if (isNaN(a)) {
+            alert('Вы ввели не число!'); continue;
+        } 
+        if (a === '' || a === null) {
+            alert('Введите число!'); continue;
+        }
+        a = Number(a);
+        }
+        while (isNaN(b) || b === '' || b === null){
```

Comment:
The code for `a` and `b` is the same. If you want to use the same code twice, you can create a function.